modules/cluster_estimation/cluster_estimation.py

```diff
@@ -159,24 +168,95 @@ def run(
         # Store new input data
         self.__current_bucket += self.__convert_detections_to_point(detections)
 
+        # Decide to run
+        if not self.__decide_to_run(run_override):
+            return False, None
+
+        # sort bucket by label in descending order
+        self.__current_bucket = self.__sort_by_labels(self.__current_bucket)
+        detections_in_world = []
+
+        # init search parameters
+        ptr = 0
+
+        # itterates through all points
+        while ptr <= len(self.__current_bucket):
+            # reference label
+            label = self.__current_bucket[ptr][2]
+
+            # creates bucket of points with the same label
+            bucket_labelled = []
+            while ptr < len(self.__current_bucket) and self.__current_bucket[ptr][2] == label:
+                bucket_labelled += [self.__current_bucket[ptr]]
+                ptr += 1
+
+            # skip if no objects have label=label
+            if len(bucket_labelled) == 0:
+                continue
+
+            check, labelled_detections_in_world = self.cluster_by_label(
```

Comment:
use the variable name `result` instead of `check`, as we use that for most of the things elsewhere